# A spatially resolved human glioblastoma atlas reveals distinct cellular and molecular patterns of anatomical niches

Glioblastoma (GBM) is a dynamic ecosystem with high plasticity where the complex interplay between different cellular components contributes to disease progression. Although single-cell RNA (scRNA)-seq has revealed remarkable cellular heterogeneity of GBM, the spatial organization of its diverse cell types is currently lacking. Here we present a comprehensive dataset of 115,914 spatial transcriptomes across 32 tissue sections of genotyped glioma samples. We present spatial maps of 56 fine-grained cellular components, including previously unrecognized oligodendrocyte subtypes and their interactive networks within each anatomical niche. Additionally, we utilized Xenium in situ technology to generate subcellular resolution spatial transcriptomic data, enhancing our understanding of GBM at a granular level. Our data provides novel insights into the cellular architecture of GBM. This valuable and openly shared resource will help develop new and effective combinatorial therapies to target multiple niches simultaneously to improve patient outcomes.

# Data exploration:
[Video showing exploration of scRNA-seq, CITE-seq, and Visium spatial transcriptome datasets](https://www.youtube.com/watch?v=wgA2_yQMOpM)

[Video showing exploration of Xenium spatial transcriptome dataset]([https://www.youtube.com/watch?v=wgA2_yQMOpM](https://www.youtube.com/watch?v=f6yDlKXgTb0))

Tools used:

RCCv2: https://github.com/nameetas/RCCv2

Data availability:

All the datasets can be downloaded or explored <a href = "https://gbmvisium.snu.ac.kr/">here</a>.

In case of any queries please contact <nameeta.shah@gmail.com>.
