# A spatially resolved human glioblastoma atlas reveals distinct cellular and molecular patterns of anatomical niches

Glioblastoma (GBM) is a highly plastic ecosystem where complex interplay between different cellular components contributes to disease progression. Although single cell RNA-seq greatly helped us to understand cellular heterogeneity of GBM, our knowledge regarding spatial organization of its cellular components is currently lacking. We performed spatial transcriptomics with matched multiomics profiling on a set of genotyped glioma samples. We present spatial maps of 10 major cellular components and their spatial interaction networks, including previously unrecognized subtypes of oligodendrocyte and stromal cell population in GBM. This work revealed that each GBM anatomical niche harbors a distinct combination of cellular components. Furthermore, deconvolution of bulk RNA-seq data using the integrated spatial and single cell atlas revealed clinically relevant GBM ecotypes. Our data provides comprehensive insights into the cellular architecture of GBM at  high spatial resolution. It will be a valuable resource to develop effective combinatorial therapies to target all tumor-fostering niches simultaneously.

Tools used:

Interactive heatmap for Ligand-Receptor interactions: https://nameetashah.shinyapps.io/Heatmap/

RCCv2: https://github.com/nameetas/RCCv2

Data availability:
Please contact <nameeta.shah@gmail.com> to access the data.

Supplementary files:

[Supplementary File 1](https://drive.google.com/file/d/1U6udzEqgecvUnRj2QAxlSQEtUiWuQCQI/view?usp=share_link)

[Supplementary File 2](https://drive.google.com/file/d/1u3Gw7MBHgDwZtvvCZyZxegjeb7D0JqEa/view?usp=share_link)
