# A spatially resolved human glioblastoma atlas reveals distinct cellular and molecular patterns of anatomical niches

Glioblastoma (GBM) is a highly aggressive brain tumor characterized by extreme cellular heterogeneity and poor clinical outcomes. Although single-cell analyses have uncovered diverse tumor and stromal cell states, their spatial architecture within the tumor microenvironment is still insufficiently defined. Here we present a multi-modal, spatially resolved atlas of human GBM by integrating spatial transcriptomics from both fresh frozen and FFPE tissue, matched single cell transcriptomic and proteomic data, and subcellular resolution in situ profiling. Using a custom 348 gene panel enriched for vascular markers, we map previously unrecognized GBM-associated endothelial, pericyte, and fibroblast states, and uncover their niche-specific associations with malignant and immune compartments. Notably, we identified a distinct oligodendrocyte subset confined to core and perivascular regions, characterized by transcriptional signatures predictive of recurrence and adverse clinical outcome. Our publicly accessible atlas provides a high-resolution framework for investigating GBM spatial biology and offers a resource for identifying spatially restricted therapeutic vulnerabilities.

The pre-print of the manuscript can be found [here](https://pmc.ncbi.nlm.nih.gov/articles/PMC10639919/).

![Figure1](https://github.com/user-attachments/assets/b381fe05-b8c9-4cdf-bdd5-a414635259ef)

#Data Availability

Raw data generated in this study will be made available upon request. Processed data and interactive browsers can be accessed at [here](https://gbmvisium.snu.ac.kr).  The single-cell and CITE-seq data can be explored [here](https://multiomic-gbm.cells.ucsc.edu/). All code and accompanying video tutorials have been deposited in GitHub (https://github.com/nameetas/TSKGA). Spatial datasets (Visium, Xenium, IHC, both our cohort and public datasets) are available at https://www.amaranth-insights.com. The user can login using the following credentials:
Email: amaranthinsights@gmail.com
Password: password@123
Additional information and resources are available from the corresponding authors upon request.



# Data exploration:

[![Slide 16_9 - 2 (1)](https://github.com/nameetas/TSKGA/assets/41729594/9a8b0036-06c5-4c63-9ae6-85de039099e5)](https://www.youtube.com/watch?v=wgA2_yQMOpM)

[Video showing exploration of scRNA-seq, CITE-seq, and Visium spatial transcriptome datasets](https://www.youtube.com/watch?v=wgA2_yQMOpM)

[Video showing exploration of Xenium spatial transcriptome dataset](https://www.youtube.com/watch?v=f6yDlKXgTb0)

Tools used:

RCCv2: https://github.com/nameetas/RCCv2

Data availability:

All the datasets can be downloaded or explored <a href = "https://gbmvisium.snu.ac.kr/">here</a>.

In case of any queries please contact <nameeta.shah@gmail.com>.
